dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/TelemetryLongRunningTest.java

```diff
@@ -0,0 +1,83 @@
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.microprofile.telemetry.internal_fat;
+
+import static com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions.SERVER_ONLY;
+
+import java.time.Duration;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.custom.junit.runner.Mode;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+import io.openliberty.microprofile.telemetry.internal_fat.apps.longrunning.LongRunningTask;
+import io.openliberty.microprofile.telemetry.internal_fat.common.spanexporter.InMemorySpanExporter;
+import io.openliberty.microprofile.telemetry.internal_fat.common.spanexporter.InMemorySpanExporterProvider;
+import io.opentelemetry.sdk.autoconfigure.spi.traces.ConfigurableSpanExporterProvider;
+
+@Mode(TestMode.FULL)
+@RunWith(FATRunner.class)
+public class TelemetryLongRunningTest extends FATServletClient {
+
+    public static final String SERVER_NAME = "Telemetry10LongRunning";
+    public static final String APP_NAME = "LongRunningApp";
+    public static final Duration DEFAULT_TIMEOUT = Duration.ofSeconds(120); // 120 seconds
+
+    @Server(SERVER_NAME)
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        WebArchive app = ShrinkWrap.create(WebArchive.class, APP_NAME + ".war")
+                        .addPackage(LongRunningTask.class.getPackage())
+                        .addPackage(InMemorySpanExporter.class.getPackage())
+                        .addAsServiceProvider(ConfigurableSpanExporterProvider.class, InMemorySpanExporterProvider.class)
+                        .addAsResource(new StringAsset("otel.sdk.disabled=false\notel.traces.exporter=in-memory"),
+                                       "META-INF/microprofile-config.properties");
+
+        ShrinkHelper.exportAppToServer(server, app, SERVER_ONLY);
+        server.startServer();
+    }
+
+    @Test
+    public void testLongLastingTask() throws Exception {
+        server.setMarkToEndOfLog();
+
+        long timeoutNanos = DEFAULT_TIMEOUT.toNanos();
+        long startNanoTime = System.nanoTime();
+        while (System.nanoTime() - startNanoTime < timeoutNanos) {
+            try {
+                Thread.sleep(1000);
+                runTest(server, APP_NAME + "/longrunning", "testLongRunningTask");
+            } catch (InterruptedException e) {
+            }
```

Comment:
Don't catch `InterruptedException` - unless you're deliberately interrupting your own task, you can assume this means that the application or server is trying to shut down (e.g. the user running the test pressed Ctrl-C).

In this case, we should fail the test, rather than attempt to keep running it.

Sometimes we'll catch `InterruptedException` to wrap it in a `RuntimeException`, but in general we shouldn't be catching and ignoring `InterruptedException`.